packages/core/src/lib/base.ts

```diff
@@ -202,8 +210,9 @@ export abstract class FrontendBase<B extends Backend, O extends Options>
    * @inheritDoc
    */
   public async setContext(nextContext: Context): Promise<void> {
-    const context = await this.getContext();
+    await this.getContext();
```

Comment:
Same here. Let's try not to operate on some internal state and always get/set state transparently to be more direct.